paddle/fluid/operators/distributed/trainer_heart_beat_monitor.cc

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/operators/distributed/trainer_heart_beat_monitor.h"
+#include <chrono>  // NOLINT
+#include <ctime>
+
+namespace paddle {
+namespace operators {
+namespace distributed {
+
+DEFINE_int32(trainer_update_interval_secs, 900,
+             " the longest time interval between the trainer update variables");
+
+inline int GetCurrentUS() {
```

Comment:
I think there is a timer in platform::Timer, could you reuse that code or improve upon that code?

Reply:
Time is designed for debugging, I can not reuse that code here.